Function trains: Amend syntax (#902)
Dyalog/documentation: programming-reference-guide/docs/introduction/trains.md

Reviewer: ```suggestion
```apl
        g h
      f g h
      A g h
```
```

Reviewer: ```suggestion
The 2-item train `(g h)` is termed an *atop* and the 3-item trains `(f g h)` and `(A g h)` are termed *forks*. To distinguish the two styles of *fork*, we can use the terms *fgh-fork* or *Agh-fork*.
```

> In particular, trains may be applied to a single array (monadic use) or between 2 arrays (dyadic use), providing  six new constructs.

Reviewer: ```suggestion
In particular, trains can be applied to a single array (monadic use) or between two arrays (dyadic use), providing six new constructs:
```

```apl
     (  g h)Y ←→         g (  h Y)   ⍝ monadic       atop
    X(  g h)Y ←→         g (X h Y)   ⍝ dyadic        atop

     (f g h)Y ←→ (  f Y) g (  h Y)   ⍝ monadic (fgh) fork
    X(f g h)Y ←→ (X f Y) g (X h Y)   ⍝ dyadic  (fgh) fork

     (A g h)Y ←→    A    g (  h Y)   ⍝ monadic (Agh) fork
    X(A g h)Y ←→    A    g (X h Y)   ⍝ dyadic  (Agh) fork

> ```

Reviewer 2: ```suggestion
|----------:|:-:|--------------------:|--------------------|
| `(  g h)Y`| ⇔ |         `g (  h Y)` | monadic atop       |
|`X(  g h)Y`| ⇔ |         `g (X h Y)` | dyadic atop        |
|           |   |                     |                    |
| `(f g h)Y`| ⇔ | `(  f Y) g (  h Y)` | monadic (fgh) fork |
|`X(f g h)Y`| ⇔ | `(X f Y) g (X h Y)` | dyadic  (fgh) fork |
|           |   |                     |                    |
| `(A g h)Y`| ⇔ |    `A    g (  h Y)` | monadic (Agh) fork |
|`X(A g h)Y`| ⇔ |    `A    g (X h Y)` | dyadic  (Agh) fork |
```

Author: As discussed, more styling is needed here, possibly with some help from @rikedyp, to hide the column/row borders and to improve cell text alignment. Otherwise, this is a good idea.

Reviewer: This is becoming much more complicated than it needs to be, and definitely should not involve changes to the CSS. I will review it properly on Monday (committed to too many other bits this afternoon!).

Reviewer: Try this (copy in from the unrendered version of this comment!):

<style id="ID">
.md-typeset #ID+div table {
  margin-left: 2em;
  & tr:nth-child(2n) td {
    padding-bottom: 1em!important;
  }
  & td {
    border: none;
	&:nth-child(2) {
	  padding: 0 1em !important;
	}
	&:last-child {
	  padding-left: 1em !important;
	}
    & code {
      white-space:pre;
    }
  } 
}
</style>

|------------:|:--:|-----------------:|------------|
|  `(  g h) Y`| ⇔ |        `g (  h Y)` | atop     |
|`X (  g h) Y`| ⇔ |        `g (X h Y)` | atop     |
|  `(f g h) Y`| ⇔ |`(  f Y) g (  h Y)` | fgh-fork |
|`X (f g h) Y`| ⇔ |`(X f Y) g (X h Y)` | fgh-fork |
|  `(A g h) Y`| ⇔ |      `A g (  h Y)` | Agh-fork |
|`X (A g h) Y`| ⇔ |      `A g (X h Y)` | Agh-fork |

Author: Looks good - thanks!
